inst/rmarkdown/templates/gigascience_article/skeleton/skeleton.Rmd

```diff
@@ -0,0 +1,548 @@
+
+## Header 2
+subsection
+
+### Header 3
+subsubsection
+
+#### Header 4
+paragraph
+
+##### Header 5
+subparagraph
+
+###### Header 6
+subsubparagraph
+
+## Blockquotes
+
+> This is a quote. Lorem ipsum dolor sit amet, 
+> consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore 
+> magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
+> laboris nisi ut aliquip ex ea commodo consequat. 
+
+## Unordered list (bullets)
+
+- This is a point in a bullet list.
+    - A sub-point must be indented four more spaces than the parent.
+        - The same is true for a sub-sub-point.
+- A third point.
+
+## Ordered list (numbered list)
+
+1. This is a point in a numbered list.
+    - A sub-point must be indented four more spaces than the parent.
+        - The same is true for a sub-sub-point.
+1. A third point.
+
+Use R to generate tables rather than attempting to specify them in rmarkdown.
+
+## Equations
+
+```
+inline equation: $A = \pi*r^{2}$
+```
+inline equation: $A = \pi*r^{2}$
+
+```
+full-line equation: $$A = \pi*r^{2}$$
+```
+$$A = \pi*r^{2}$$
+
+## Tables
+
+### Making tables with rmarkdown
+
+Use R to generate your rmarkdown tables, e.g., see Table \ref{tab:markdown}.  Granted, literal rendering of rmarkdown tables is not working, but even if it worked, it would makes no sense to use it when you can use R to lay out tables more coherently.
+
+\begin{table}
+\caption{Table generated using rmarkdown}
+\label{tab:markdown}
+```{r setup, echo=FALSE, warning=FALSE}
+library(knitr)
+knitr::opts_chunk$set(echo = TRUE)
+
+kable(
+  t(
+    data.frame(
+      header=c("Right","Left","Center")
+    , value=c(1,2,3) 
+    )
+  )
+, align = c("r","l","c")
+, format = "latex")
+```
+\end{table}
+
+### Laying out tables in \LaTeX{}
+
+Table layout in \LaTeX{} is much less intuitive than using R, but this might be your only choice unless you define a section that has only one column per page.
+
+
+
+## Pre-existing figure
+
+It is inadvisable to generate images from R code chunks, but it's fine to include pre-existing images.
+
+```{r warning=FALSE, dev='pdf', fig.show='hide', echo=FALSE, message=FALSE}
+# That being said, a file is being generated here solely for demonstration purposes.
+invisible({
+  pdf(file=file.path(getwd(),'pressure.pdf'))
+  plot(pressure)
+  dev.off()
+})
+```
+
+### Including a figure using \LaTeX{}
+
+The easier way to include a figure is with \LaTeX{}.
+Figure \ref{fig:pressure} is included this way, and the figure number is a clickable reference.
+```
+% "bt!" means favor putting the table at the 
+%       top or bottom of the column, if possible
+% "b!"  means favor putting the table at the 
+%       bottom of the column, if possible
+% "t!"  means favor putting the table at the 
+%       top of the column, if possible
+\begin{figure}[bt!]
+\centering
+\includegraphics[width=\linewidth]{pressure.pdf}
+\caption{Pressure figure from \LaTeX{}}
+\label{fig:pressure}
+\end{figure}
+```
```

Comment:
This could be just 3 lines instead of 13:

````
```{r pressure, fig.caption="Pressure figure from \\LaTeX{}", fig.pos="bt!", fig.align="center"}
plot(pressure)
```
````

It will generate the figure label `fig:pressure.`

Reply:
The title for this section is "Including a figure using \LaTeX{}".  Yes, it does take 13 lines.  Yes, I do have a lot to learn. Yes, maybe I can learn how to do figures effectively using rmarkdown. But I still want to document how to do it with LaTeX, too.

I tried the following
```
{r pressureFromMarkdown, fig.caption="Pressure figure from markdown", fig.pos="bt!", fig.align="center", fig.width=1, fig.height=1, echo=FALSE, dpi=576}
knitr::include_graphics('pressure.pdf')
```
but the figure does not scale to the width of the column (instead, it's the full width of the page).  It seems to ignore fig.width and DPI.

Until I learn another way, I have to resort to LaTeX.

Reply:
Okay, let's move on and ignore my suggestion here.